app/Repositories/Summit/DoctrineMemberRepository.php

```diff
@@ -212,4 +688,74 @@ public function getByEmailExclusiveLock($email): ?Member
             ->setLockMode(\Doctrine\DBAL\LockMode::PESSIMISTIC_WRITE)
             ->getOneOrNullResult();
     }
+
+    /**
+     * @inheritDoc
+     */
+    public function getSubmittersBySummit(Summit $summit, PagingInfo $paging_info, Filter $filter = null, Order $order = null)
+    {
+        return $this->getParametrizedAllByPage(function () use ($summit) {
+            return $this->getEntityManager()->createQueryBuilder()
+                ->distinct(true)
+                ->select("e")
+                ->from($this->getBaseEntity(), "e")
+                // we need to have SIZE(e.presentations) > 0 OR SIZE(e.moderated_presentations) > 0 for a particular summit
+                ->where(" 
+                         EXISTS (
+                            SELECT __p.id FROM models\summit\Presentation __p 
+                            JOIN __p.speakers __spk WITH __spk.member = e.id 
+                            JOIN __p.created_by __cb93
+                            WHERE __p.summit = :summit
+                         ) OR
+                         EXISTS (
+                            SELECT __p1.id FROM models\summit\Presentation __p1 
+                            JOIN __p1.moderator __md WITH __md.member = e.id 
+                            JOIN __p1.created_by __cb94
+                            WHERE __p1.summit = :summit
+                          )")
+                ->setParameter("summit", $summit);
+        },
+            $paging_info,
+            $filter,
+            $order,
+            function ($query) {
+                //default order
+                return $query->addOrderBy("e.id", 'ASC');
+            });
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getSubmittersIdsBySummit(Summit $summit, PagingInfo $paging_info, Filter $filter = null, Order $order = null)
+    {
+        return $this->getParametrizedAllIdsByPage(function () use ($summit) {
+            return $this->getEntityManager()->createQueryBuilder()
+                ->distinct(true)
+                ->select("e.id")
+                ->from($this->getBaseEntity(), "e")
+                // we need to have SIZE(e.presentations) > 0 OR SIZE(e.moderated_presentations) > 0 for a particular summit
+                ->where(" 
+                         EXISTS (
+                            SELECT __p.id FROM models\summit\Presentation __p 
+                            JOIN __p.speakers __spk WITH __spk.member = e.id 
+                            JOIN __p.created_by __cb93
+                            WHERE __p.summit = :summit
+                         ) OR
+                         EXISTS (
+                            SELECT __p1.id FROM models\summit\Presentation __p1 
+                            JOIN __p1.moderator __md WITH __md.member = e.id 
+                            JOIN __p1.created_by __cb94
+                            WHERE __p1.summit = :summit
+                          )")
```

Comment:
this is not needed
u need to join by creator , remove moderator
creator is submitter

Reply:
u only need one single jhoin

Reply:
removed